web/src/message_view.ts

```diff
@@ -97,8 +97,12 @@ export function reset_ui_state(opts: {trigger?: string}): void {
     compose_banner.clear_message_sent_banners(true, skip_automatic_new_visibility_policy_banner);
 }
 
-export function changehash(newhash: string, trigger: string): void {
-    if (browser_history.state.changing_hash) {
+export function changehash(
+    newhash: string,
+    trigger: string,
+    remove_current_hash_from_history = false,
+): void {
+    if (browser_history.state.changing_hash || remove_current_hash_from_history) {
```

Comment:
I don't quite understand the implications of this change. It looks like you're keeping the non-`near` view in the history. That seems potentially nice for the URL bar. But does it mean that navigation to another view and then hitting the "Back" button results in your marking everything above the current scroll position as read? (The alternative would be to go through the `window.history.replaceState` code path, so the `/near/` URL is what stays in history)

This kind of detail makes me wonder... We could in theory make some other state change with the effect of setting `can_mark_messages_read=false`.

Reply:
Sorry, I didn't push the updated commit here. Please review this commit again.

We are replacing the current narrow which doesn't have `near` term and with the new one which has `near` term.

Added code assertions to clarify that.